subsys/net/lib/utils/addr_utils.c

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2019 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <net/socket.h>
+
+/* These utility functions are intended to be on top of POSIX API, and don't
+ * make sense if it's not available.
+ */
+#ifdef CONFIG_NET_SOCKETS_POSIX_NAMES
+
+const char *net_addr_str_find_port(const char *addr_str)
+{
+	const char *p = strrchr(addr_str, ':');
+
+	if (p == NULL) {
+		return NULL;
+	}
+
+	/* If it's not IPv6 numeric notation, we guaranteedly got a port */
+	if (*addr_str != '[') {
+		return p + 1;
+	}
+
+	/* IPv6 numeric address, and ':' preceeded by ']' */
+	if (p[-1] == ']') {
+		return p + 1;
+	}
+
+	/* Otherwise, just raw IPv6 address, ':' is component separator */
+	return NULL;
+}
+
+int net_getaddrinfo_addr_str(const char *addr_str, const char *def_port,
+			     const struct addrinfo *hints,
+			     struct addrinfo **res)
+{
+	const char *port = net_addr_str_find_port(addr_str);
```

Comment:
> If we are calling this from user app, then we need to have more null checks for the input parameters.

Well, why? What's so special about NULL? What we'd be interested in catching *any* invalid value, but we can't, because user can write `char *foo` and get uninitiated var with value from stack. In this regard, NULL is just one of billions invalid value. If user passes NULL, the app will crush with a nice stacktrace (if built with suitable options), and we check for NULL, it doesn't crash there, but crashes elsewhere, or just silently misbehaves, in which regard checking for NULL (out of all other possible invalid values) is more trouble than boon.

Reply:
Also, isn't checking argument one of those things the "Zephyr way" goes against? From a security perspective, this isn't really the best place to be checking values, and the ones that matter are those that come from external to the system.

Reply:
As this function is to be callable from user space, it makes sense to validate the input.

Reply:
> As this function is to be callable from user space, it makes sense to validate the input.

It makes sense, but we can't really do that, as explained above, just can pretend to do that. All those NULL checks just add code size, but surprisingly, efficiency still seems to be a concern for Zephyr:  https://github.com/zephyrproject-rtos/zephyr/pull/15969.

 So, I'll add it here, but we'll end up needing to add CONFIG_DISABLE_NULL_CHECKS to disable all those useless checks sprinkled around and get the code size back.

Reply:
> As this function is to be callable from user space

And this argument... These functions are themselves userspace, they are just application convenience functions, factored out for reuse. It's not like that they're kernel-level functions, params to which should be validated (much more thoroughly than NULL checks).

Anyway, added NULL check.